src/backend/distributed/commands/table.c

```diff
@@ -71,53 +75,97 @@ PreprocessDropTableStmt(Node *node, const char *queryString)
 
 	Assert(dropTableStatement->removeType == OBJECT_TABLE);
 
+	List *citusTableOids = ExtractCitusTableOidsFromDropStmt(dropTableStatement);
+
+	if (list_length(citusTableOids) <= 0)
+	{
+		/* no citus tables are to be dropped by DROP command, simply return */
+		return NIL;
+	}
+
+	/* detach partitions of tables in MX nodes if any */
+	DetachPartitionsFromMXNodes(citusTableOids);
+
+	return NIL;
+}
+
+
+/*
+ * ExtractCitusTableOidsFromDropStmt returns OIDs of the citus tables to be
+ * dropped in a DROP TABLE command.
+ */
+static List *
+ExtractCitusTableOidsFromDropStmt(DropStmt *dropTableStatement)
+{
+	/* this function is only used for DROP TABLE commands currently */
+	Assert(IsA(dropTableStatement, DropStmt) && dropTableStatement->removeType ==
+		   OBJECT_TABLE);
+
+	List *citusTableOids = NIL;
+
 	List *tableNameList = NULL;
 	foreach_ptr(tableNameList, dropTableStatement->objects)
 	{
 		RangeVar *tableRangeVar = makeRangeVarFromNameList(tableNameList);
-		bool missingOK = true;
 
+		const bool missingOK = true;
 		Oid relationId = RangeVarGetRelid(tableRangeVar, AccessShareLock, missingOK);
 
-		/* we're not interested in non-valid, non-distributed relations */
-		if (relationId == InvalidOid || !IsCitusTable(relationId))
+		if (!OidIsValid(relationId) || !IsCitusTable(relationId))
```

Comment:
Here it is relationId, a few lines later it is relationOid, I think we should use the same format for both.